Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -262,6 +262,55 @@ class BrowserViewController: UIViewController {
                 self.tabManager.addTabAndSelect(request, isPrivate: PrivateBrowsingManager.shared.isPrivateBrowsing)
             }
         }
+        
+        backgroundDataSource.initializeFavorites = { sites in
+            DispatchQueue.main.async {
+                defer { Preferences.NewTabPage.preloadedFavoritiesInitialized.value = true }
+                
+                if Preferences.NewTabPage.preloadedFavoritiesInitialized.value
```

Comment:
These types of checks are outside the async block below. I do not think they need to be on main thread, but regardless, implementations should probably be consistent.

Reply:
`Bookmark.hasFavorites` has to happen on main thread, otherwise it fails for some reason, even though that method uses view context under the hood

I could only move the preference check outside of async block if you want